src/segment_reader.rs

```diff
@@ -342,6 +346,157 @@ impl AsyncSegmentReaderImpl {
     }
 }
 
+/// A wrapper around the AsyncSegmentReader.
+///
+/// It maintains a buffer and prefetches data to fill that buffer in the background so that
+/// every read call can read from the local buffer instead of waiting for server response.
+pub(crate) struct AsyncSegmentReaderWrapper {
```

Comment:
The name `AsyncSegmentReaderWrapper` implies that it's a generic wrapper of `AsyncSegmentReader` while it is a wrapper, it provides some additional functionality, it should be named accordingly. Like maybe `BufferedAsyncSegmentReader` or `PrefetchingAsyncSegmentReader`.

Reply:
fixed, thanks!